Chapter10/pretty_print.erl

```diff
@@ -12,111 +12,104 @@ index() -> index(get_path()).
 
 % Exercise 10-2: Indexing Revisited
 index(File) ->
-	ets:new(indexTable, [ordered_set, named_table, {keypos, 1}]),
-	processFile(File),
-	prettyIndex(),
-	ets:delete(indexTable).
+  ets:new(indexTable, [ordered_set, named_table, {keypos, 1}]),
+  processFile(File),
+  prettyIndex(),
+  ets:delete(indexTable).
 
 
 processFile(File) ->
-	{ok,IoDevice} = file:open(File,[read]),
-	processLines(IoDevice,1).
+  {ok,IoDevice} = file:open(File,[read]),
+  processLines(IoDevice,1).
 
 processLines(IoDevice,N) ->
-	case io:get_line(IoDevice,"") of
-		eof -> ok;
-		Line ->
-			processLine(Line,N),
-			processLines(IoDevice,N+1)
-	end.
+  case io:get_line(IoDevice, "") of
+    eof -> ok;
+    Line ->
+      processLine(Line, N),
+      processLines(IoDevice, N+1)
+  end.
 
 
 -define(Punctuation, "[\\ |\\,|\\.|\\;|\\:|\\t|\\n|\\(|\\)]").
 
 processLine(Line,N) ->
-	Words = re:split(Line, ?Punctuation, [{return,list}]),
-	processWords(Words, N).
-
+  Words = re:split(Line, ?Punctuation, [{return,list}]),
+  processWords(Words, N).
 
 processWords(Words,N) ->
-	case Words of
-		[] -> ok;
-		[Word|Rest] ->
-			if
-				length(Word) > 3 ->
-					Normalise = string:to_lower(Word),
-					ets:insert(indexTable, {{Normalise , N}} );
-				true -> ok
-			end,
-			processWords(Rest,N)
-	end.
-
+  case Words of
```

Comment:
```erlang
processWords([], _N) ->
    ok;
processWords([Word | Rest], N) when length(Word) > 3 ->
    % ...
processWords([_Word | Rest], N) ->
    processWords(Rest, N).
```

Reply:
This function was given from the chapter not written by me, but I've updated it anyway to work as you've shwon